webflow-hacks/form-hacks/11.ts

```diff
@@ -0,0 +1,20 @@
+// when the DOM is ready
+document.addEventListener("DOMContentLoaded", function() { 
+  // get & store the page url in a variable
+  const url = location.href;	
+  // set the page url as the hidden input's value
+  (document.querySelector('#insertpageurl') as HTMLInputElement).value = url;
```

Comment:
Please don't do type assertions, this can lead to runtime checks as we can't guarantee that these elements will always exist on the page.